### PR TITLE
Fixed #78 and #73

### LINKED ISSUE #78: Tests Fail On Clean Checkout
VmRuntimeJettyAuthTest.testAuth_UntrustedInboundIp:154 expected:<307> but was:<403>
  VmRuntimeJettyAuthTest.testAuth_UntrustedInboundIpWithQuery:165 expected:<307> but was:<403>

### PR DESCRIPTION
**Pull request by gregw**
Cleaned up poms to clearly use java 1.8
Use 127.0.0.2 as force remote IP, which is known to be untrusted.

### CODE REVIEW

**Comment by gregw** [comment]
fix for #78 and #73

File: appengine-managed-runtime/src/main/java/com/google/apphosting/vmruntime/VmRequestUtils.java [hunk 1]
```diff
@@ -38,6 +38,9 @@ public static boolean isTrustedRemoteAddr(boolean isDevMode, String remoteAddr)
       return true;
     } else if (remoteAddr.startsWith(LINK_LOCAL_IP_NETWORK)) {
       return true;
+    } else if (remoteAddr.equals("127.0.0.2")) {
```
Comment (jboynes) [inline appengine-managed-runtime/src/main/java/com/google/apphosting/vmruntime/VmRequestUtils.java]: Curious about the semantics required here. [RFC5735](https://tools.ietf.org/html/rfc5735) says:
>127.0.0.0/8 - This block is assigned for use as the Internet host loopback address.  A datagram sent by a higher-level protocol to an address anywhere within this block loops back inside the host.  This    is ordinarily implemented using only 127.0.0.1/32 for loopback.  As   described in [RFC1122], Section 3.2.1.3, addresses within the entire   127.0.0.0/8 block do not legitimately appear on any network anywhere.

Which means we should either trust all "127." addresses in the block, or only the "ordinary implementation" of "127.0.0.1" but treating just 127.0.0.2 as special seems odd.

**Comment by ludoch** [comment]
LGTM

**Comment by jboynes** [comment]
@ludoch thoughts on trusted ips ?

**Comment by ludoch** [comment]
Please, add your concerns.

**Comment by jboynes** [comment]
@ludo My [prior comment](https://github.com/GoogleCloudPlatform/appengine-java-vm-runtime/pull/86#discussion_r39869695) above about making 127.0.0.2 special

**Comment by gregw** [comment]
@jboynes The previous impl was using the hosts local address and assuming that would not be trusted.  Yet it had a devenv setup that trusted all localhosts.

Thus I just added a single exclusion to mark 127.0.0.2 as untrusted and left the 127.0.0. as trusted.
I could not see any other 127 addresses being used, so it should be find to change this to just trust 127.0.0.1, but that is essentially equivalent to what we have now.

**Pull request merged**